### DISCUSSION TITLE
[Support] 请问项目/组件中能否获取当前正在运行的 ip/port 等信息？ | Is it possible to get project's ip or port on the fly?

### DISCUSSION THREAD

**Discussion by ldc-37** [midwayjs/midway#3577] [Q&A] [answered]
### Describe the problem（描述问题）

场景：写了一个组件，会动态增加路由到主项目中，期望能够在控制台输出这些路由的访问地址，并且可以直接点击打开链接。这就意味着需要拿到 protocol, host & port，但文档中似乎没有发现有 api 暴露这些信息。

尝试找了一下 cli-plugin-dev 里边的确有准确的**实际使用**的 port，但来源于子进程的 message 事件且似乎没有对外暴露。
<img width="540" alt="image" src="https://github.com/midwayjs/midway/assets/34739463/5faa18b5-783b-45b1-8ab9-93255440ee48">

另外 egg.js 有[相关讨论](https://github.com/eggjs/egg/issues/2652)，我不确定 midway 是如何把配置传递给 egg 的，是否有更简便的方法能拿到这些信息呢？

### Midway Versions（Midway 版本）

3.12.3

**Comment by czy88840616** [comment]
端口取 process.env.MIDWAY_HTTP_PORT 即可。

**Reply by ldc-37** [reply]
感谢回复，但我测试了一下，只有通过命令行参数的方式传入 port，这里才能取到 port，否则全程都是 undefined

<img width="550" alt="image" src="https://github.com/midwayjs/midway/assets/34739463/770596a4-b1ae-4857-a0d6-8c03bd54e878">
<img width="461" alt="image" src="https://github.com/midwayjs/midway/assets/34739463/50b88444-29bc-4d81-890e-5410b56fad31">


同样在 cli-plugin-dev 中还有个环境变量 MIDWAY_LOCAL_DEV_PORT，它看起来至少在开发环境保存了实际使用的 port，然而该变量被赋值的时间晚于 onReady 和 onServerReady，也不太好用。
<img width="596" alt="image" src="https://github.com/midwayjs/midway/assets/34739463/886edf15-6530-4a5d-81b8-273ad0dfd94c">

**Reply by czy88840616** [reply, marked as answer]
取配置中的ip和环境变量或一下

**Reply by ldc-37** [reply]
我也想到了这个方案，看起来只能这样了。只是在端口冲突时显示的就不是实际使用的端口了。

备选方案：或许可以循环等待一下 MIDWAY_LOCAL_DEV_PORT 有值

**Reply by czy88840616** [reply]
另外，cli 已经废弃了，就不需要在 cli 中改了吧。

**Reply by ldc-37** [reply]
ok，感谢提醒。后边有空改成 mwtsc
